global/scripts/debug.shlib

```diff
@@ -19,10 +18,25 @@ then
     source "$HCPPIPEDIR/global/scripts/log.shlib" "$@"
 fi
 
+if [ -z ${HCP_DEBUG_USETRAP} ] ; then
```

Comment:
We could just call the `debug_enable_trap` functions, so that we don't have nother thing referencing the specific magic variable names.

How do you plan to enable coloring in QuNex, by exporting the variables in advance?  If so, you'll need the current -z tests on the color/verbose settings, but the "use trap" and "abort" settings don't need conditionals, since QuNex will use the same defaults.

Reply:
So, I think the debug_disable_trap and debug_enable_trap should be completely under the control of the scripts themselves, because for a section that uses a custom message, the expected pattern is disable, command+message, enable, and if the command is itself a pipeline script, then with the conditionals and the variable flagged for export, it will inadvertently run the second script in no-trap mode.

On the flip side, if you plan to use export to control the coloring/verbosity, it isn't clear that the scripts should ever call the functions to enable or disable color or verbosity (as that would override the QuNex environment).  Thus, the functions for color and verbosity only seem to be useful if we intend to add coloring and verbose option switches to the pipelines.

Reply:
In other words, I think the default setting for "use trap" and "abort on error" should be unconditional.

Reply:
If the defaults are unconditional, then how can be set to something else? The point is to be able to set the environment to one's preference and not overwrite it by default. 

Yes, the coloring and verbosity enabling and disabling functions are there, if a script would need for some reason to change them. But I am ok in removing them completely so that they do not ever change from the environment setting.

Reply:
Only the defaults for the behavior of the debug trap should be unconditional.  Users should not ever mess with the logical effects of the debug trap, because that changes the results of running the script.  The scripts should be in sole control of these things, otherwise weird things can happen.  It is a bad idea for users to be able to set the script to not exit without editing the script (the fact that the exporting by itself would have additional unintended side effects just makes the situation worse).

Changing whether the scripts print more messages or print them in color doesn't affect where the scripts exit or what outputs they make, so those can be user preferences.  I do not feel strongly about whether the functions to set them should be removed.

Reply:
I see your point. Fixed it.